modules/csharp/examples/code_snippets/Program.cs

```diff
@@ -2602,4 +2602,72 @@ public void InitReplication()
 
         // end::sgw-act-rep-initialize[]
     }
+
+    // tag::custom-logger
+    internal class MyCoolCustomLogger : ILogger
+    {
+        public LogLevel Level { get; set; }
+
+        public void Log(LogLevel level, LogDomain domain, string message)
+        {
+            // Do something cool with this information
+        }
+    }
+    // end::custom-logger
+
+    // tag::custom-log-sink
+    internal class MyCoolLogSink : BaseLogSink
+    {
+        public MyCoolLogSink(LogLevel level)
+            : base(level)
+        {
+
+        }
+
+        protected override void WriteLog(LogLevel level, LogDomain domain, string message)
+        {
+            // Do something cool with this information
+        }
+    }
+    // end::custom-log-sink
+
+    public void OldLoggingApi()
+    {
+        // tag::console-logging[]
+        Database.Log.Console.Level = LogLevel.Verbose;
+        // end::console-logging[]
+
+        // tag::file-logging[]
+        Database.Log.File.Config = new LogFileConfiguration("path/to/log/directory")
+        {
+            MaxRotateCount = 2, // Save 3 log files (i.e. 2 rotated and 1 current)
+            MaxSize = 1024 * 512, // 512KB per file, then rotated
+        };
+
+        Database.Log.File.Level = LogLevel.Verbose;
+        // end::file-logging[]
+
+        // tag::custom-logging[]
+        Database.Log.Custom = new MyCoolCustomLogger() { Level = LogLevel.Verbose };
+        // end::custom-logging[]
+    }
+
+    public void NewLoggingApi()
+    {
+        // tag::new-console-logging[]
+        LogSinks.Console = new ConsoleLogSink(LogLevel.Verbose);
+        // end::new-console-logging[]
+
+        // tag::new-file-logging[]
+        LogSinks.File = new FileLogSink(LogLevel.Verbose, "path/to/log/directory")
+        {
+            MaxKeptFiles = 3, // Save 3 log files (i.e. 2 rotated and 1 current)
```

Comment:
I haven't checked the .NET implementation. But isn't this supposed to be the same value for the user (2)? I know we accounted for the -/+ 1 when passing the param around (at least I do for iOS).

Reply:
Handling the offset by one from core happens underneath the API but it doesn’t make sense to name the property “max kept files” and not have it describe how many files exist. You can check the defaults which are 1 for the old api and 2 for the new.